src/org/akvo/resumed.clj

```diff
@@ -120,26 +120,52 @@
             (DatatypeConverter/parseBase64Binary)
             (String.))))))
 
+(defn get-host
+  "Returns the HOST for a given request
+  It attempts to honor: X-Forwared-Host, Origin, Host headers"
+  [req]
+  (let [host (or (get-header req "x-forwarded-host")
+                 (get-header req "origin")
+                 (first (.split (or (get-header req "host") "") ":")))]
+    (if (str/blank? host)
+      (:server-name req)
```

Comment:
Maybe we can move this to the last thing in the "or" in line 127, if like 129 returns a nil, which it could do if we used a regular expression to extract the host, or `some->`